contracts/token/RarePizzasBox.sol

```diff
@@ -114,68 +127,61 @@ contract RarePizzasBox is
     }
 
-    // Member Functions
-
-    /**
-     * Get the current bitcoin price in wei
-     */
-    function getBitcoinPriceInWei() public view returns (int256) {
-        return bitcoinPriceInWei;
-    }
+    // IRarePizzasBoxAdmin
 
-    /**
-     * allows the contract owner to mint up to a specific number of boxes
-     */
-    function mint(address to, uint8 count) public virtual onlyOwner {
-        require(count > 0, 'RAREPIZZA: must provide a count');
+    function mint(address toPizzaiolo, uint8 count) public virtual override onlyOwner {
+        require(toPizzaiolo != address(0), 'RAREPIZZA: dont be silly');
+        require(count > 0, 'RAREPIZZA: need a number');
 
-        require(totalSupply().add(count) <= maxSupply(), 'RAREPIZZA: mint would exceed maxSupply');
+        require(totalSupply().add(count) <= maxSupply(), 'RAREPIZZA: exceeds supply.');
         require(
             _minted_pizza_count.current().add(count) <= MAX_MINTABLE_SUPPLY,
             'RAREPIZZA: mint would exceed MAX_MINTABLE_SUPPLY'
         );
 
         for (uint256 i = 0; i < count; i++) {
             _minted_pizza_count.increment();
-            uint256 id = _getNextPizzaTokenId();
-            _safeMint(to, id);
-            _assignBoxArtwork(id);
+            _internalMintWithArtwork(toPizzaiolo);
         }
     }
 
-    /**
-     * allows owner to purchase to a specific address
-     */
-    function purchaseTo(address to) public payable virtual onlyOwner {
-        require(totalSupply().add(1) <= MAX_TOKEN_SUPPLY, 'RAREPIZZA: purchase would exceed maxSupply');
-        require(to != msg.sender, 'RAREPIZZA: Thats how capos get whacked');
+    function purchaseTo(address toPaisano) public payable virtual override onlyOwner {
+        require(toPaisano != address(0), 'RAREPIZZA: dont be silly');
+        require(totalSupply().add(1) <= MAX_TOKEN_SUPPLY, 'RAREPIZZA: exceeds supply.');
+        require(toPaisano != msg.sender, 'RAREPIZZA: Thats how capos get whacked');
 
         uint256 price = getPrice();
-        require(price == msg.value, 'RAREPIZZA: price must be on the curve');
+        require(msg.value >= price, 'RAREPIZZA: price too low');
         _purchased_pizza_count.increment();
-        uint256 id = _getNextPizzaTokenId();
-        _safeMint(to, id);
-        _assignBoxArtwork(id);
+        _internalMintWithArtwork(toPaisano);
     }
 
-    /**
-     * allows the owner to update the cached bitcoin price
-     */
-    function updateBitcoinPriceInWei(int256 fallbackValue) public virtual onlyOwner {
+    function setPresaleAllowed(uint8 count, address[] memory toPaisanos) public virtual override onlyOwner {
+        for (uint256 i = 0; i < toPaisanos.length; i++) {
+            require(toPaisanos[i] != address(0), 'RAREPIZZA: dont be silly');
+            _presaleAllowed[toPaisanos[i]] = count;
+        }
+    }
+
+    function setSaleStartTimestamp(uint256 epochSeconds) public virtual override onlyOwner {
```

Comment:
This could be used to halt sales and allow whitelist people to buy only

Reply:
yes correct that is the desired behavior